### PR TITLE
Kf 856 gh109 fix resolve CVE

### PR DESCRIPTION
**Pull request by i-chvets**
### Design
For more details refer to: https://github.com/canonical/notebook-operators/issues/109

Ignore new files under `./kubeflow/`, but please review all update files, eg. `Dockerfiles`, `Makefiles`, `requirements.*`

This PR has limited updates. It downgrades builder images to previous base OS images.

### Testing

- Follow instructions in `./image-definitions/README.md` to create a VM and install all required tools
- Checkout this PR in the VM (`main` branch already contains bulk of the changes for the https://github.com/canonical/notebook-operators/issues/109)
- Follow instructions in README.md to build and scan updated images.
  - No need to do setup part, because it is only used for creating updates.
  - No need to specify tag, it will be setup automatically.
- Expected results:
  - Rebuilt updated images.
  - Trivy reports.

### CODE REVIEW

File: image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # --- Build the backend kubeflow-wheel ---
-FROM python:3.7-slim-bullseye AS backend-kubeflow-wheel
+FROM python:3.7-slim-buster AS backend-kubeflow-wheel
```
Comment (i-chvets) [inline image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile:2]: Downgrading to previous version (not fixing builder images).

File: image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 WORKDIR /src
 
 COPY ./common/backend/ .
 RUN python3 setup.py bdist_wheel
 
 # --- Build the frontend kubeflow library ---
-FROM node:18-bullseye-slim as frontend-kubeflow-lib
+FROM node:12-buster-slim as frontend-kubeflow-lib
```
Comment (i-chvets) [inline image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile:10]: Downgrading to previous version (not fixing builder images).

File: image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile [hunk 3]
```diff
@@ -22,7 +22,7 @@ COPY ./common/frontend/kubeflow-common-lib/tsconfig.json .
 RUN npm run build
 
 # --- Build the frontend ---
-FROM node:18-bullseye-slim as frontend
+FROM node:12-buster-slim as frontend
```
Comment (i-chvets) [inline image-definitions/kubeflow/components/crud-web-apps/jupyter/Dockerfile:26]: Downgrading to previous version (not fixing builder images).

**Comment by ca-scribner** [comment]
images build successfully with following results:

Image build artifacts:

```
REPOSITORY               TAG                 IMAGE ID       CREATED                  SIZE
notebook-controller      v1.6.1              b8c2e07ea149   Less than a second ago   73.3MB
<none>                   <none>              c99c62cbcd79   4 seconds ago            2.11GB
jupyter-web-app          v1.6.1              fcf6afa50fba   3 minutes ago            184MB
<none>                   <none>              8618ab06e647   3 minutes ago            1.42GB
<none>                   <none>              3c526fcb3dca   8 minutes ago            693MB
<none>                   <none>              246955bcdd52   10 minutes ago           119MB
<none>                   <none>              c15ff77a7b46   10 minutes ago           436MB
base                     v1.6.1              8d686b9ef267   10 minutes ago           436MB
ubuntu                   20.04               f32fe8df6a4c   5 days ago               72.8MB
gcr.io/distroless/base   debug               3713eb4249de   6 days ago               21.5MB
python                   3.7-slim-buster     b39f6ea4df78   7 days ago               116MB
python                   3.7-slim-bullseye   7ef53f615a8b   7 days ago               123MB
golang                   1.19                8295ad49803f   2 weeks ago              993MB
node                     12-buster-slim      44f66a685237   9 months ago             155MB
```

Scan results:

Tag: v1.6.1
Date: 2023.1.31
CVEs per image:
 IMAGE | BASE | CRITICAL | HIGH | MEDIUM | LOW 
 -- | -- | -- | -- | -- | -- 
 notebook-controller:v1.6.1 | debian:11.6 | 0 | 6 | 2 | 0 
 jupyter-web-app:v1.6.1 | debian:11.6 | 0 | 1 | 0 | 0 
 base:v1.6.1 | ubuntu:20.04 | 0 | 0 | 0 | 0 
 ubuntu:20.04 | ubuntu:20.04 | 0 | 0 | 0 | 0 
 gcr.io/distroless/base:debug | debian:11.6 | 0 | 0 | 0 | 0 
 python:3.7-slim-buster | debian:10.13 | 1 | 1 | 0 | 0 
 python:3.7-slim-bullseye | debian:11.6 | 0 | 1 | 0 | 0 
 golang:1.19 | debian:11.6 | 7 | 12 | 11 | 0 
 node:12-buster-slim | debian:10.12 | 13 | 21 | 11 | 5 
| | Totals: | 21 | 42 | 24 | 5

**Review by ca-scribner — APPROVED** [review]

**Pull request merged**